modules/xmpp/JingleSessionPC.js

```diff
@@ -2640,12 +2658,12 @@ export default class JingleSessionPC extends JingleSession {
         // context a common object for one run of ssrc update (source-add and source-remove) so we can match them if we
         // need to
         const ctx = {};
+        const removedSsrcInfo = getSignaledSourceInfo(sdpDiffer);
 
-        if (removedAnySSRCs) {
-            const sourceInfo = getSignaledSourceInfo(sdpDiffer);
-
+        if (removedAnySSRCs && removedSsrcInfo.ssrcs.length) {
```

Comment:
why would `removedAnySSRCs` be true while `removedSsrcInfo.ssrcs.length` is `0`?

Reply:
This would happen in p2p when a remote peer adds a second video source. A recv-only m-line will be created as a result in the local SDP without any SSRC.

Reply:
you're writing about mline being added while the if condition is about any sources being removed

Reply:
I don't understand how an mline being added/removed affects the two conditions specifically

Reply:
Sorry was AFK and had typed the original response on my mobile. This comment is meant for source-add case. When the remote peer adds a new source, a new m-line in the remote description is created and therefore a recv-only m-line without any SSRC is added to the local SDP as a result. Therefore on renegotiation, this code detects that there is a new m-line and SDPDiffer.toJIngle returns true because of the new m-line even though there is no new SSRC. I didn't change the existing variables and therefore it may be confusing here. This happens only in the case of P2P. For JVB case, the interop.toPlanB() in the getLocalDescription getter takes care of ignoring the recv-only m-lines.  
I added this extra check for the source-remove just in case to make sure source-remove with no SSRCs is ever signaled to the peer.

Reply:
maybe the condition then should be based only on `removedSsrcInfo.ssrcs.length`? it is confusing to have two sources of truth